### PR TITLE
Allow for CSI provisioned volume or storage class as source

### PR DESCRIPTION
**Pull request by emosbaugh**
I understand it is not the intention of the project as clearly stated in the readme, but I found with this small change it is possible to migrate from CSI to CSI volumes.

Our use case is migrating from a flex driver storage class, to a temporary csi storage class, deleting the flex storage class, and migrating to a csi storage class with the same name as the original flex storage class.

### CODE REVIEW

File: pkg/migration/migration.go [hunk 1]
```diff
@@ -78,19 +78,20 @@ func migratePVC(client *k8s.Clientset, pvc v1.PersistentVolumeClaim, destination
 	}
 	logger.DefaultLog("PV found %q ", pv.Name)
 
-	logger.DefaultLog("Update Reclaim policy from Delete to Reclaim for PV: %s", pv.Name)
-	err = k8sutil.UpdateReclaimPolicy(client, pv)
-	if err != nil {
-		return fmt.Errorf("failed to update ReclaimPolicy for PV object %s: %v", pvc.Spec.VolumeName, err)
-	}
-
+	// retrieve and validate the volume name before starting the migration
```
Comment (emosbaugh) [inline pkg/migration/migration.go:81]: i've reordered this to prevent validation issues from mutating the resource

**Review by Madhu-1 — COMMENTED** [review]
@emosbaugh changes LGTM, update the readme for this support?

**Comment by emosbaugh** [comment]
> @emosbaugh changes LGTM, update the readme for this support?

added documentation to the readme

**Comment by subhamkrai** [comment]
let's wait for #43 before merging

**Review by Madhu-1 — APPROVED** [review]

**Comment by subhamkrai** [comment]
@emosbaugh can you please just rebase on top of main, we just fix the ci after that we'll good to merge the pr

**Comment by emosbaugh** [comment]
> @emosbaugh can you please just rebase on top of main, we just fix the ci after that we'll good to merge the pr

rebased

**Pull request merged**